mirrord/operator/src/crd/session.rs

```diff
@@ -90,15 +251,37 @@ pub struct SessionCopyTarget {
 }
 
 /// Status of a mirrord session.
+#[derive(Clone, Debug, Deserialize, Eq, PartialEq, Serialize, JsonSchema)]
+#[serde(rename_all = "camelCase")]
+pub struct MirrordClusterSessionAgent {
+    /// Agent connection info to target's agents.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub connection_info: Option<AgentKubernetesConnectInfo>,
+    /// If the agent is of the ephemeral variaty
+    pub ephemeral: bool,
+    /// Agent spawn error if there is one.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub error: Option<ErrorMessage>,
+    /// The phase of agent's pod.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub phase: Option<String>,
```

Comment:
We could have a pod phase enum I guess. Up to you though, not important.

Reply:
I would have wanted to make one but since the `Pod`'s status stores it as String I think I should keep it the same